chirpstack-concentratord-sx1302/src/config/mod.rs

```diff
@@ -321,6 +327,13 @@ pub fn get(filenames: Vec<String>) -> Configuration {
 
     let mut config: Configuration = toml::from_str(&content).expect("Error parsing config file");
 
+    // decode gateway id
+    let bytes = hex::decode(&config.gateway.gateway_id).expect("Could not decode gateway_id");
```

Comment:
This should be something like:

```rust
if let Some(gateway_id) = &config.gateway.gateway_id {
    // hex decode gateway_id
   // cast Vec<u8> = [u8; 8]
   // set gateway_id_bytes to Some(gateway_id)
}
```

Reply:
With `gateway_id` as a `String` and not a `Option<String>`, this pattern will always match so the `if let` is useless. So changed `gateway_id` to `Option<String>`. This way, `gateway_id` can be commented out in the toml file and as such the semtech EUI is used. When `gateway_id` is used, it will check for correct length and use this EUI instead.